packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -100,38 +105,141 @@ typedef WidgetTesterCallback = Future<void> Function(WidgetTester widgetTester);
 /// ```
 @isTest
 void testWidgets(
-  String description,
-  WidgetTesterCallback callback, {
-  bool skip = false,
-  test_package.Timeout timeout,
-  Duration initialTimeout,
-  bool semanticsEnabled = true,
-}) {
+    String description,
+    WidgetTesterCallback callback, {
+      bool skip = false,
+      test_package.Timeout timeout,
+      Duration initialTimeout,
+      bool semanticsEnabled = true,
+      Iterable<TestDimension<dynamic>> matrix = const <TestDimension<dynamic>>[],
```

Comment:
you could just make this have the default value `[DefaultDimension()]`, which would avoid the special casing for empty necessary below (you could just assert non-empty)